cmd/tunneld/tunneld.go

```diff
@@ -164,9 +169,36 @@ func startAutocert(opts *options, server *tunnel.Server, logger log.Logger) {
 		"addr", "80,443",
 	)
 	go func() {
-		fatal("failed to start HTTP: %s", http.ListenAndServe(":http", m.HTTPHandler(server)))
+		fatal("failed to start HTTP: %s", http.ListenAndServe(httpAddr, m.HTTPHandler(server)))
 	}()
 	go func() {
 		fatal("failed to start HTTPS: %s", s.ListenAndServeTLS("", ""))
 	}()
 }
+
+func hostPolicy(server *tunnel.Server) autocert.HostPolicy {
+	return func(_ context.Context, host string) error {
+		_, _, subscribed := server.Subscriber(host)
+		if !subscribed {
+			return fmt.Errorf("acme/autocert: host `%s` not subscribed", host)
+		}
+		return nil
+	}
+}
+
+func trimPort(hostPort string) (host string) {
+	host, _, _ = net.SplitHostPort(hostPort)
+	if host == "" {
+		host = hostPort
+	}
+	return
+}
+
+func cacheDir(letsEncryptCacheDir string) string {
+	const certs = "autocerts"
+	filepath.Join(os.Getenv("HOME"), ".cache", certs)
+	if letsEncryptCacheDir != "" {
```

Comment:
Why join, user must provide a proper final directory.

Reply:
yep that shouldn't be there only `filepath.Join(letsEncryptCacheDir, certs)`

Reply:
It would be good if all tunnel related files would be in one place so try setting the default to `~/.tunnel/certs`?